### PR TITLE
Amplitude detuning procedure

### PR DESCRIPTION
**Pull request by JoschD**

### CODE REVIEW

File: docs/measurements/procedures/ampdet.md [hunk 1]
```diff
@@ -3,44 +3,364 @@
...
+          <figure>
+              <img src="../../../assets/images/amplitude_detuning_procedure/tune_settings.png">
+              <figcaption>Tune settings.</figcaption>
+          </figure>
+          </details>
+    - [ ] <details class="nodeco"><summary>Good Frequency Resolution (`TurnBits > 15`)</summary>
+          <p>
+          As the tune shift can be very small, we would want a good resolution in frequency, which can be controlled by `TurnBits`.
+          The standard value of `TurnBits` of `20` (leading to 2<sup>20</sup> complex coefficients, i.e. 2<sup>21</sup> spectral lines) is a good start, but can lead to excessive memory use when analysing 15&minus;20 files at once. 
+          I estimate (from experience), that with `19` turn-bits, you will need 60GB to 80GB of RAM, `20` will obviously double and `18` half that value.
```
Comment (fsoubelet) [inline docs/measurements/procedures/ampdet.md:213]: halve

File: docs/measurements/procedures/ampdet.md [hunk 2]
```diff
@@ -3,44 +3,364 @@
...
+        </p></details>
+- [ ] <details class="nodeco"><summary>Run combined Optics Analysis</summary>
+    <p>
+    After all measurements are satisfyingly cleaned, the optics analysis (`Get Optics`) can be run on **all measurements combined**, so that they end up all
+    in the same folder in the same `Kick File` (per plane).
+    Give the analysis a nice name, identifying what kind of measurement you are doing (e.g. `b1_ampdet_vertical_30cm_with_xing`).
+    </p></details>
+- [ ] <details class="nodeco"><summary>Final Cleaning in Kick-Files</summary>
+    <p>
+    After the optics analysis has run, the only input from it that is needed for the detuning analysis, is are the `Kick-Files`.
```
Comment (fsoubelet) [inline docs/measurements/procedures/ampdet.md:313]: is are

File: docs/measurements/procedures/ampdet.md [hunk 3]
```diff
@@ -3,44 +3,364 @@
...
+        To account for any tune drifts of the machine during the measurement, the tune data is corrected by the data from the BBQ.
+        As this data is very noisy (and contains the outliers from the AC-Dipole excitation), it needs to be cleaned first.
+        A moving average of the given `Window length` is used determining a tune estimation over the kick timerange.
+        This data is extracted from timber and saved into a `bbq_ampdet.tfs` file in the optics results folder. 
+        So in case you run the same detuning analysis again, but maybe want to modify the BBQ cleaning parameters, this data can be re-used by ticking `Use previous BBQ`.
+        The first cleaning method `outliers` is using the same algorithm as `Auto Clean`, and requires an `outlier limit` to be set. 
+        BBQ data deviating less than this limit from the moving average are never cleaned. 
+        The moving average is then calculated on the cleaned data and the closest value to the kick time subtracted from the natural tune measured during the kick.
+        The second method `cut` let's use choose a fixed `cut` around the assumed natural tunes and data beyond that value is cleaned.
```
Comment (fsoubelet) [inline docs/measurements/procedures/ampdet.md:351]: lets

File: docs/measurements/procedures/ampdet.md [hunk 4]
```diff
@@ -3,44 +3,364 @@
...
+        In this case also `Fine windowing` can be enabled, which performs another moving average on the now cleaned data and datapoints deviating more than the `fine cut` 
+        value are also cleaned.<br>
+        </p></details>
+
+??? info "Quick TurnBit Estimation"
+    Let's assume a small detuning of 10 &middot; 10<sup>3</sup> m<sup>&minus;1</sup> and an action increase between kicks of 0.020 &mu;m / 20 kicks,
+    leading to a detuning difference between two kicks of &Delta;Q = 1 &middot; 10<sup>&minus;5</sup>.
+    The frequency resolution $\Delta f$ is the inverse of the length of the turns-data (halfed, as we have pairs), so 2<sup>&minus;TurnBits&minus;1</sup> :<br>
```
Comment (fsoubelet) [inline docs/measurements/procedures/ampdet.md:359]: halved

**Review by fsoubelet — COMMENTED** [review]
I'm glad I made a workflow to upload the site build in PRs.

Many small comments / suggestions / spotted typos.
Very nice work, I'm eager to merge this once comments are resolved!

Also, I learned things :D

File: docs/measurements/procedures/ampdet.md [hunk 5]
```diff
@@ -3,44 +3,364 @@
...
+          When kicking with crossing: head the warning above! Losses may occur with only a small amplitude increase!
+          </p></details>
+    - [ ] <details class="nodeco"><summary>Check for tune drift</summary>
+          <p>
+          If you are kicking with reduced tune deltas, it is also important to have an eye on the tune drift of the machine, 
+          so that you do not further decrease the distance between natural tune and excitation. 
+          Update the tune in multiturn if necessary.
+          The tune drifts will be compensated in the python analysis step by using data from the BBQ.
+          </p></details>
+    - [ ] Try to reach 0.015&mu;m - 0.020&mu;m in action ($2J_x$)
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:97]: well, no. It's the other way around. I use the value given here in the box below to make an estimate

File: docs/measurements/procedures/ampdet.md [hunk 6]
```diff
@@ -3,44 +3,364 @@
...
+          When kicking with crossing: head the warning above! Losses may occur with only a small amplitude increase!
+          </p></details>
+    - [ ] <details class="nodeco"><summary>Check for tune drift</summary>
+          <p>
+          If you are kicking with reduced tune deltas, it is also important to have an eye on the tune drift of the machine, 
+          so that you do not further decrease the distance between natural tune and excitation. 
+          Update the tune in multiturn if necessary.
+          The tune drifts will be compensated in the python analysis step by using data from the BBQ.
+          </p></details>
+    - [ ] Try to reach 0.015&mu;m - 0.020&mu;m in action ($2J_y$)
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:119]: idem

File: docs/measurements/procedures/ampdet.md [hunk 7]
```diff
@@ -3,44 +3,364 @@
...
+          When kicking with crossing: head the warning above! Losses may occur with only a small amplitude increase!
+          </p></details>
+    - [ ] <details class="nodeco"><summary>Check for tune drift</summary>
+          <p>
+          If you are kicking with reduced tune deltas, it is also important to have an eye on the tune drift of the machine, 
+          so that you do not further decrease the distance between natural tune and excitation. 
+          Update the tune in Multiturn if necessary.
+          The tune drifts will be compensated in the python analysis step by using data from the BBQ.
+          </p></details>
+    - [ ] Try to reach 0.015&mu;m - 0.020&mu;m in action in both planes
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:141]: idem

File: docs/measurements/procedures/ampdet.md [hunk 8]
```diff
@@ -3,44 +3,364 @@
...
+        in the LHC at $\beta^*$ = 30cm up to &plusmn;170&mu;rad IP1-V/IP5-H crossing
+        and &plusmn;150&mu;rad IP1-H, &minus;150&mu;rad to +140&mu;rad IP5-V separation.
+        </p></details>
+
 - [ ] <details class="nodeco"><summary>Correct Coupling</summary>
-      <p>Kick at medium amplitude and wait for the result.
-      Then put the result into the coupling correction tool.
+      <p>
+      As always, coupling should be well corrected to $|C-| \leq 10^{-3}$.
+      This can be easily achieved by performing diagonal kicks of mediocre strength, and get the correction values from the GUI.<br>
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:34]: fixed

File: docs/measurements/procedures/ampdet.md [hunk 9]
```diff
@@ -3,44 +3,364 @@
...
+        <figure class="rightFigure">
+            <img class="clickImg" src="../../../assets/images/amplitude_detuning_procedure/FrequencyChart_both_tunes_found.png">
+            <figcaption>Good Spectrum.</figcaption>
+        </figure>
+        </details>
+    - [ ] Repeat until the spectrum looks usable
+
+!!! info "Tune Deltas"
+
+    in the past, clean amplitude detuning measurements could be achieved with tune deltas (signs as given in Multiturn) of <br>
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:62]: fixed

File: docs/measurements/procedures/ampdet.md [hunk 10]
```diff
@@ -3,44 +3,364 @@
...
+    The action can be calculated if $\beta$ and peak-to-peak is known for a specific BPM by $2J = \frac{(0.5 \cdot pk2pk)^2}{\beta}$.
+    Otherwise it can always be checked running optics analysis on a file and looking into the Optics &rarr; Action/Tune tab or the kick-file.
 
 ## Analysis
 
-- [ ] Load data
-- [ ] Check data for correct tunes
-- [ ] Analyse data
-- [ ] Run BBQ correction
+Analysis performed with the Beta-Beat GUI `omc3-v0.1.1`, python package `omc3` `v0.4.0`.
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:151]: added more info

File: docs/measurements/procedures/ampdet.md [hunk 11]
```diff
@@ -3,44 +3,364 @@
...
+      Make sure you are loading the correct data and check the logbook for misfired kicks etc.
+      </p></details>
+- [ ] <details class="nodeco"><summary>Run Spectral Analysis</summary>
+      <p>
+      A bad spectral analysis can be recovered by the steps mentioned in "Cleaning", 
+      but a good frequency spectrum and well found natural tunes will save you a lot of time later on.
+      </p></details>
+    - [ ] <details class="nodeco"><summary>Small `tolerance` (&asymp;10<sup>&minus;3</sup>)</summary>
+          <p>
+          The tune tolerance (found in the `Tune Settings`) specifies in what region around the assumed natural tune (see next step, the region is $f$ &plusmn; tolerace), the highest frequency line is chosen to be the natural tune.
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:172]: fixed

File: docs/measurements/procedures/ampdet.md [hunk 12]
```diff
@@ -3,44 +3,364 @@
...
+          To not accidentally capture the main tune or other excited resonances close by, the tolerance should be kept low (&asymp;10<sup>&minus;3</sup>).
+          In case of large detuning (e.g. 40 &middot; 10<sup>3</sup> m<sup>&minus;1</sup> &times; 0.016m = 6.4 &middot; 10<sup>&minus;3</sup>) or change of the AC-Dipole frequency (e.g. to adapt for tune drifts), 
+          this could mean that the natural tune will not fall into this window anymore.
+          The both can be avoided using `Autotunes` and maybe adapting the tune deltas (see next step),
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:176]: fixed

File: docs/measurements/procedures/ampdet.md (hunk 1 above)
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md:213]: fixed

File: docs/measurements/procedures/ampdet.md [hunk 13]
```diff
@@ -3,44 +3,364 @@
...
+        **"LOAD COARSE SETTINGS FOR NLO AT 30 CM"** is the current (run 3, 2022) collimation sequence for AC-Dipole kicks 
+        in the LHC at $\beta^*$ = 30cm up to &plusmn;170&mu;rad IP1-V/IP5-H crossing
+        and &plusmn;150&mu;rad IP1-H, &minus;150&mu;rad to +140&mu;rad IP5-V separation.
+        </p></details>
+
 - [ ] <details class="nodeco"><summary>Correct Coupling</summary>
-      <p>Kick at medium amplitude and wait for the result.
-      Then put the result into the coupling correction tool.
+      <p>
+      As always, coupling should be well corrected to $|C-| \leq 10^{-3}$.
```
[the remark this answers is not in the document]
Comment (JoschD) [reply docs/measurements/procedures/ampdet.md]: changed to \approx

File: docs/measurements/procedures/ampdet.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fsoubelet) [reply docs/measurements/procedures/ampdet.md:97]: Got it

File: docs/measurements/procedures/ampdet.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (fsoubelet) [reply docs/measurements/procedures/ampdet.md]: Looks good

**Pull request merged**